flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/jsonplan/JsonPlanGenerator.java

```diff
@@ -117,10 +120,25 @@ public static String generatePlan(
                 // write the core properties
                 JobVertexID vertexID = vertex.getID();
                 int storeParallelism = vertexParallelism.getParallelism(vertexID);
+                int parallelism =
+                        storeParallelism != -1 ? storeParallelism : vertex.getParallelism();
```

Comment:
I think the `-1` here can also be replaced by `ExecutionConfig#PARALLELISM_DEFAULT`

Reply:
ok